pkg/collect/run.go

```diff
@@ -155,12 +155,12 @@ func runPod(ctx context.Context, client *kubernetes.Clientset, runCollector *tro
 		},
 	}
 
-	if runCollector.ImagePullSecret != nil && runCollector.ImagePullSecret.Name != "" {
-		err := createSecret(ctx, client, pod.Namespace, runCollector.ImagePullSecret)
+	if runCollector.ImagePullSecret != nil && runCollector.ImagePullSecret.Data != nil {
```

Comment:
The name check here caused the issue. Generated secrets don't have names. They get names generated by k8s.